packages/camera/android/src/test/java/io/flutter/plugins/camera/FilePathValidatorTest.java

```diff
@@ -0,0 +1,97 @@
+package io.flutter.plugins.camera;
+
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertFalse;
+import static junit.framework.TestCase.assertTrue;
+import static org.mockito.Mockito.*;
+
+import java.io.File;
+import org.junit.Test;
+
+public class FilePathValidatorTest {
+
+  @Test
+  public void validate_ReturnsIsDirectory() {
+    FilePathValidator.FileFactory mockFactory = mock(FilePathValidator.FileFactory.class);
+    File mockFile = mock(File.class);
+    String filePath = "foo";
+
+    when(mockFile.exists()).thenReturn(true);
+    when(mockFile.isDirectory()).thenReturn(true);
+    when(mockFactory.makeFile(filePath)).thenReturn(mockFile);
+
+    FilePathValidator validator = new FilePathValidator(filePath, mockFactory);
+    validator.validate();
+
+    assertEquals(
+        "filePathInvalid",
+        "File at path '"
```

Comment:
nit:
It's typically preferred to include "foo" in the String instead of concatenating the variable in tests.